### PR TITLE
Compiler warnings

### PR DESCRIPTION
**Pull request by tombentley**
### Type of change

- Refactoring

### Description

This PR fixes all the compiler warnings, excluding tests and the code generated by sundrio.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java [hunk 1]
```diff
@@ -2093,7 +2099,7 @@ int getPodIndexFromPvcName(String pvcName)  {
                     List<String> maybeDeletePvcs = res.result().stream().map(pvc -> pvc.getMetadata().getName()).collect(Collectors.toList());
                     List<String> desiredPvcs = zkCluster.generatePersistentVolumeClaims().stream().map(pvc -> pvc.getMetadata().getName()).collect(Collectors.toList());
 
-                    persistentClaimDeletion(maybeDeletePvcs, desiredPvcs).setHandler(futureResult.completer());
+                    persistentClaimDeletion(maybeDeletePvcs, desiredPvcs).setHandler(futureResult);
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java]: Why are you changing this? Out of curiosity, if it works I'm fine with the change.

Comment (tombentley) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java]: `.completer()` is now deprecated. A `Future` is it's own completer (i.e. implements `Handler<AsyncResult<T>>`).

Comment (scholzj) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java]: Thanks

**Comment by tombentley** [comment]
@strimzi-ci run tests

**Comment by tombentley** [comment]
@Frawless Jenkins finished without test results. It is misconfigured?

**Comment by tombentley** [comment]
@strimzi-ci run tests

**Comment by tombentley** [comment]
@strimzi-ci run tests

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/TopicOperator.java [hunk 2]
```diff
@@ -96,7 +96,7 @@ protected TopicOperator(String namespace, String cluster, Labels labels) {
 
         super(namespace, cluster, labels);
         this.name = topicOperatorName(cluster);
-        this.replicas = TopicOperatorSpec.DEFAULT_REPLICAS;
+        this.replicas = io.strimzi.api.kafka.model.TopicOperatorSpec.DEFAULT_REPLICAS;
```
Comment (ppatierno) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/TopicOperator.java]: Why the entire package is needed? Does the compiler complain about that?

**Comment by tombentley** [comment]
@strimzi-ci run tests

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/TopicOperator.java (hunk 2 above)
Comment (tombentley) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/TopicOperator.java]: You cannot annotate an `import` stmt with `@SuppressWarnings("deprecation")`, so those warnings will be emitted and the build will fail. Work around is to not use `import` and fully qualify the class everywhere you need it.

**Pull request merged**